packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -47,6 +48,11 @@ export default createRule<
         type: 'object',
         additionalProperties: false,
         properties: {
+          allowIfStatements: {
+            type: 'boolean',
+            description:
+              'Allow autofix to optional chain on if statements with a single call expression in the consequent.',
```

Comment:
[Docs] This option name reads to me like it's allowlisting `if` statements. As in, turning this on tells the rule to ignore them. The opposite of what it does! Going off the current functionality, I think it'd be more appropriately named as `checkIfStatements`.

However, long-term I think we'll want to have the checking of `if` statements on by default. I expect most folks will prefer to have the rule check them as well, and only a minority will want to opt out using the option.

Proposal: let's have the option's functionality be inverted so that it's an opt-out for `if` statements. And, with that, a slightly more precise name:

```suggestion
          ignoreIfStatements: {
```

```suggestion
          allowIfStatements: {
            type: 'boolean',
            description:
              'Whether to ignore `if` statements with a single expression in the consequent.',
```